### PR TITLE
recreate missing files

### PR DESCRIPTION
**Pull request by cloudmagic80**
## Summary

*This work is behind a feature toggle (flipper): YES
This PR fixes a bug in our program. In a very small sample of users, an IVC CHAMPVS form user will occasionally get a 500 error message and will not able to submit the complete form. This is caused by the temporary multi-stamp file being deleted in the kubernetes pod before the vets-api pdf filler completes. This PR will allow a retry of the multi stamp creation if the file is missing.

## Related issue(s)

https://github.com/department-of-veterans-affairs/vets-api/pull/19006

https://github.com/department-of-veterans-affairs/vets-api/pull/19662

User story:
https://github.com/department-of-veterans-affairs/va.gov-team/issues/97231

## Testing done

- [x ] *New code is covered by unit tests*

- *If this work is behind a flipper:* YES
  - *Tests need to be written for both the flipper on and flipper off scenarios. [Docs](https://depo-platform-documentation.scrollhelp.site/developer-docs/feature-toggles-guide#Featuretogglesguide-Backendexample).*

## What areas of the site does it impact?
All IVC CHAMPVA FORMS

## Unit test 
added

### CODE REVIEW

**Review by stevelong00 — COMMENTED** [review]
Looks good, just needs unit tests.

**Review by michaelclement — APPROVED** [review]

File: modules/ivc_champva/app/controllers/ivc_champva/v1/uploads_controller.rb [hunk 1]
```diff
@@ -49,21 +49,51 @@ def submit_supporting_documents
...
-        end
+          statuses = Array(statuses)
 
-        [statuses, error_message]
+          # Retry attempt if specific error message is found
+          if statuses.any? do |status|
+            status.is_a?(String) && status.include?('No such file or directory @ rb_sysopen')
+          end
+            file_paths, metadata = get_file_paths_and_metadata(parsed_form_data)
+            statuses, error_message = FileUploader.new(form_id, metadata, file_paths, true).handle_uploads
```
Comment (stevelong00) [inline modules/ivc_champva/app/controllers/ivc_champva/v1/uploads_controller.rb:90]: This line is where the retry happens.  If you comment it out, the unit test that checks for the retry doesn't fail.

File: modules/ivc_champva/spec/requests/ivc_champva/v1/forms/uploads_spec.rb [hunk 2]
```diff
@@ -293,15 +293,34 @@
...
+
+        context 'when a document is loaded and is missing' do
+          before do
+            allow(file_uploader).to receive(:handle_uploads).and_return([['No such file '],
+                                                                         'File not found'])
+          end
+
+          it 'retries the file uploads and returns the error message' do
+            allow(file_uploader).to receive(:handle_uploads).and_return([[200], nil])
+
```
Comment (stevelong00) [inline modules/ivc_champva/spec/requests/ivc_champva/v1/forms/uploads_spec.rb:320]: To check that the retry actually happens I suggest adding something like `expect(file_uploader).to receive(:handle_uploads).twice`

File: modules/ivc_champva/spec/requests/ivc_champva/v1/forms/uploads_spec.rb [hunk 3]
```diff
@@ -293,15 +293,74 @@
           end
         end
 
-        context 'when file uploads fail with other errors' do
+        context 'when file uploads fail with other errors retry once' do
```
Comment (stevelong00) [inline modules/ivc_champva/spec/requests/ivc_champva/v1/forms/uploads_spec.rb:296]: Please wrap this and the following test in a `context "Feature feature_flag=false"` block like in [this example](https://depo-platform-documentation.scrollhelp.site/developer-docs/feature-toggles-guide#FeatureTogglesGuide-Backendexample) so it's obvious that these tests can be removed when we take out the feature flag.

Comment (cloudmagic80) [reply modules/ivc_champva/spec/requests/ivc_champva/v1/forms/uploads_spec.rb:296]: Not needed  because you want to test with the feature in one error

Comment (cloudmagic80) [reply modules/ivc_champva/spec/requests/ivc_champva/v1/forms/uploads_spec.rb:296]: In all situations with flipper on or off.

**Review by stevelong00 — DISMISSED** [review]
Looks good!

File: modules/ivc_champva/app/controllers/ivc_champva/v1/uploads_controller.rb [hunk 4]
```diff
@@ -49,21 +49,51 @@ def submit_supporting_documents
...
+          begin
+            file_paths, metadata = get_file_paths_and_metadata(parsed_form_data)
+            file_uploader = FileUploader.new(form_id, metadata, file_paths, true)
+            statuses, error_message = file_uploader.handle_uploads
+          rescue => e
+            attempt += 1
+            error_message_downcase = e.message.downcase
+            Rails.logger.error "Error handling file uploads (attempt #{attempt}): #{e.message}"
+
+            if (error_message_downcase.include?('failed to generate stamped file') || error_message_downcase.include?('unable to find file')) && attempt <= max_attempts
```
Comment (ericboehs) [inline modules/ivc_champva/app/controllers/ivc_champva/v1/uploads_controller.rb:67]: Could you break this line up?

**Review by michaelclement — APPROVED** [review]

**Review by ericboehs — APPROVED** [review]

**Pull request merged**